spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -80,6 +80,8 @@
 
 	private volatile Runnable stopCallback;
 
+	private volatile boolean readFromLeft = false;
```

Comment:
Semantics:

push/pop == stack
head/tail == queue

Since we've already coerced the Redis native stack semantics to queue semantics (via the adapter names - `queue-*-adapter` I can see an argument to use `enqueue = head/tail` and `dequeue=tail/head`.

That said, I think I still lean towards `pop=right/left`, `push=left/right` because it would be clearer to people who know Redis.